tools/src/wyvern/target/corewyvernIL/expression/New.java

```diff
@@ -87,8 +88,12 @@ public ValueType typeCheck(TypeContext ctx) {
 
 		TypeContext thisCtx = ctx.extend(selfName, getExprType());
 
+		boolean isResource = false;
 		for (Declaration d : decls_ExceptDelegate()) {
 			DeclType dt = d.typeCheck(ctx, thisCtx);
+			if (d instanceof DefDeclaration && ((DefDeclaration)d).isResource()) {
```

Comment:
Avoid instanceof.  Instead, rename isResource to capturesResource and make it an element of Declaration, not just DefDeclaration.